### PR TITLE
Scrape examples

### PR DESCRIPTION
**Pull request by Vrixyz**
Implements https://github.com/dimforge/parry/issues/233

Builds on top of https://github.com/dimforge/parry/pull/241

### CODE REVIEW

File: crates/parry2d/Cargo.toml [hunk 1]
```diff
@@ -68,4 +85,95 @@ simba = { version = "0.9", default-features = false }
...
 ptree = "0.4.0"
 rand = { version = "0.8" }
-macroquad = "0.4"
\ No newline at end of file
+macroquad = "0.4"
+
+[package.metadata.docs.rs]
+rustdoc-args = ["-Zunstable-options", "--cfg", "docsrs"]
+all-features = true
+cargo-args = ["-Zunstable-options", "-Zrustdoc-scrape-examples"]
```
Comment (waywardmonkeys) [inline crates/parry2d/Cargo.toml:93]: You don’t need to pass a cfg, docs.rs does that already.

I don’t think you can use all-features due to dim2/dim3 and f32/f64

you can do this like I do elsewhere:

```

[package.metadata.docs.rs]
rustdoc-args = [ "-Zunstable-options", "--generate-link-to-definition"]
cargo-args = ["-Zunstable-options", "-Zrustdoc-scrape-examples"]
```

File: crates/parry3d/Cargo.toml [hunk 2]
```diff
@@ -69,3 +81,109 @@ rstar = "0.12.0"
 oorandom = "11"
 ptree = "0.4.0"
 rand = { version = "0.8" }
+
+[package.metadata.docs.rs]
```
Comment (waywardmonkeys) [inline crates/parry3d/Cargo.toml:85]: As above…

**Review by waywardmonkeys — CHANGES_REQUESTED** [review]

File: crates/parry2d/Cargo.toml [hunk 3]
```diff
@@ -68,4 +85,95 @@ simba = { version = "0.9", default-features = false }
...
+
+[[example]]
+name = "aabb2d"
+path = "examples/aabb2d.rs"
+doc-scrape-examples = true
```
Comment (Vrixyz) [inline crates/parry2d/Cargo.toml:98]: Add a comment to explain why all examples are listed (rustdoc scrape examples)

+ check if rustc has planned to facilitate this configuration

**Review by sebcrozet — APPROVED** [review]
🎉

**Review by waywardmonkeys — APPROVED** [review]

**Pull request merged**